### PR TITLE
initial trial on adding referenced projects, need improvments

### LINKED ISSUE #40: Improving the research project page by adding Eclipse project dependencies
Dear Web Team,

We would like to improve the default template for the research project pages to include not only the Eclipse projects that the research contributes to but also the Eclipse projects that the research project uses. This will allow us to promote our projects and demonstrate that the research is (intensely) using Eclipse technologies.

the APPSTACLE is a great example using 7 Eclipse projects and we have a couple of research project which asked to be able to list used Eclipse technologies.

I have attached a possible layout as a suggestion. Your suggestions are of course more than welcome.

This is NOT an urgent request

Thanks a lot for your help

Kind regards
new-research-project-page

### PR DESCRIPTION
**Pull request by flora8984461**
Fix #40 

For now I only add referenced projects in `/project/appstacle/ ` to have a test. I do not what other referenced projects are for other projects.

Signed-off-by: Yi Liu <yi.liu@eclipse-foundation.org>

### CODE REVIEW

File: layouts/project/single.html [hunk 1]
```diff
@@ -2,29 +2,26 @@
...
-			    </a>
-			  </h3>
+
+	{{ if .Params.referenced_eclipse_projects }}
+		{{ if gt (len .Params.referenced_eclipse_projects) 0 }}
+			<h2 class="margin-top-30 margin-bottom-20">Referenced Eclipse projects</h2>
+			<div class="row">
+			{{ range .Params.referenced_eclipse_projects }}
+			    {{range first 1 (where $.Site.Data.eclipsefdn_projects "project_id" .) }}
+			    <div class="col-md-4 background-white margin-bottom-20 match-height-item vertical-align margin-right-20">
```
Comment (autumnfound) [inline layouts/project/single.html]: This should not be creating a new well for every logo. We should be displaying this content much like we are doing  for the Eclipse Project Contributions section

**Review by autumnfound — CHANGES_REQUESTED** [review]

File: layouts/project/single.html (hunk 1 above)
Comment (flora8984461) [reply layouts/project/single.html]: By well do you mean the white background?

Comment (autumnfound) [reply layouts/project/single.html]: Sort of. I mean the individual white boxes. These logos should be in one larger box that spans the width of the available area, and be an inline list (like the other logos later in the page).

**Comment by phkrief** [comment]
Looks good !

**Comment by phkrief** [comment]
Actually, my only concern is the size of the Eclipse Project logos. On some project such as APPSTACLE, they look huge (check out the KUKSA logo).
It doesn't look consistent with the other logo like the research project logo.
Thx

File: layouts/project/single.html [hunk 2]
```diff
@@ -100,6 +97,23 @@ <h2>Funded by</h2>
...
 		{{ end }}
-  	</div>
+		{{ if gt (len .Params.eclipse_projects) 0 }}
+		<div class="sidebar-panel margin-top-20 margin-bottom-20">
+			<h2>Eclipse Projects Contributions</h2>
+			<ul class="padding-left-0">
+				{{ range .Params.eclipse_projects }}
+				    {{range first 1 (where $.Site.Data.eclipsefdn_projects "project_id" .) }}
+						<li class="vertical-align">
+							<a href="{{ .url }}" class="vertical-align flex-column">
```
Comment (autumnfound) [inline layouts/project/single.html]: The class here doesn't need to be vertical-align flex-column, we can just use text-center

**Comment by flora8984461** [comment]
I added more referenced projects into some other projects to have a test. 

I am not quite sure if making the logo image on the right smaller would be better? 

@autumnfound  I am trying to use `inline list` to make it flexible to expand or not according to the contents.

Such as: 
https://deploy-preview-46--eclipsefdn-research.netlify.app/project/amalthea4public/
https://deploy-preview-46--eclipsefdn-research.netlify.app/project/pdp4e/


> Actually, my only concern is the size of the Eclipse Project logos. On some project such as APPSTACLE, they look huge (check out the KUKSA logo).
> It doesn't look consistent with the other logo like the research project logo.
> Thx

File: layouts/project/single.html [hunk 3]
```diff
@@ -2,29 +2,25 @@
...
-				  {{ .name }}
-			    </a>
-			  </h3>
+	{{ if .Params.referenced_eclipse_projects }}
+		{{ if gt (len .Params.referenced_eclipse_projects) 0 }}
+			<h2 class="margin-top-30 margin-bottom-20">Referenced Eclipse projects</h2>
+			<div class="row">
+			{{ range .Params.referenced_eclipse_projects }}
+				{{range first 1 (where $.Site.Data.eclipsefdn_projects "project_id" .) }}
+				<div class="col-md-4 background-white margin-bottom-20 match-height-item vertical-align margin-right-20">
```
Comment (autumnfound) [inline layouts/project/single.html]: This should be col-sm-4 at the most. Md means that up to ~900px there is one tile per row which doesn't look quite right

File: layouts/project/single.html [hunk 4]
```diff
@@ -2,29 +2,25 @@
...
-		      <h3>
-		        <a href="{{ .url }}">
-			      <img class="img img-responsive margin-auto padding-bottom-20 " src="{{ .logo }}" title="{{ .name }} logo"/>
-				  {{ .name }}
-			    </a>
-			  </h3>
+	{{ if .Params.referenced_eclipse_projects }}
+		{{ if gt (len .Params.referenced_eclipse_projects) 0 }}
+			<h2 class="margin-top-30 margin-bottom-20">Referenced Eclipse projects</h2>
+			<div class="row">
```
Comment (autumnfound) [inline layouts/project/single.html]: If you use a row, there should be a wrapping col around all of these cols. With the current look and feel, the boxes are countersunk 15px left (due to the row) and on screens > 1000px width, the box is against the edge of the screen.

The solution here is to likely remove the row (as it isn't needed here) or wrap the entire "Eclipse Projects" section in a col-xs-24 div.

**Review by autumnfound — CHANGES_REQUESTED** [review]

File: layouts/project/single.html (hunk 4 above)
Comment (autumnfound) [reply layouts/project/single.html]: The reason that the boxes are too far countersunk is that there are 2 rows in a row without a column class between them

File: layouts/project/single.html (hunk 3 above)
Comment (flora8984461) [reply layouts/project/single.html]: But it is a little bit weird that between 700-900px, the referenced projects logos are small, but the logos in sidebar are big. Do you think I should adjust them or just leave it like this? 🤔

image

File: layouts/project/single.html (hunk 4 above)
Comment (flora8984461) [reply layouts/project/single.html]: removed row

File: layouts/project/single.html (hunk 3 above)
Comment (autumnfound) [reply layouts/project/single.html]: You could up it to col-sm-6 if you like. The point was more that we needed the small columns covered as well as medium at least.

**Pull request merged**